pymilvus/orm/partition.py

```diff
@@ -29,9 +28,13 @@
 class Partition:
     def __init__(self, collection, name, description="", **kwargs):
         from .collection import Collection
-        if not isinstance(collection, Collection):
-            raise CollectionNotExistException(message=ExceptionsMessage.CollectionType)
-        self._collection = collection
+        if isinstance(collection, Collection):
+            self._collection = collection
+        elif isinstance(collection, str):
+            self._collection = Collection(collection)
+        else:
+            raise ValueError("Collection must be od type pymilvus.Collection or a string.")
```

Comment:
Great optimize!

Please use MilvusException here

Reply:
@XuanYang-cn 
Should we check for the case of empty string ""

Reply:
Collection could take care of the checks